src/upgrades/mongodb_upgrade.py

```diff
@@ -109,6 +121,30 @@ def _on_upgrade_charm(self, _):
             # `upgrade_resumed`
             self._reconcile_upgrade()
 
+    def _on_pre_upgrade_check_action(self, event: ActionEvent) -> None:
+        if not self.charm.unit.is_leader():
+            message = f"Must run action on leader unit. (e.g. `juju run {self.app.name}/leader {upgrade.PRECHECK_ACTION_NAME}`)"
+            logger.debug(f"Pre-upgrade check event failed: {message}")
+            event.fail(message)
+            return
+        if not self._upgrade or self._upgrade.in_progress:
```

Comment:
I think these 2 should have distinct messages. The first one effectively implies that the charm is not ready for an upgrade, correct?